rs-matter/src/utils/writebuf.rs

```diff
@@ -28,13 +28,19 @@ pub struct WriteBuf<'a> {
 
 impl<'a> WriteBuf<'a> {
     pub fn new(buf: &'a mut [u8]) -> Self {
+        let _buf_size = buf.len();
```

Comment:
delete this line?

Reply:
overall why do we need buf-size if we have a fat pointer in buf? does self.but not have a len itself anyway?

Reply:
This is old code, but TL;DR is: look at `shrink` and `expand`.